main.go

```diff
@@ -24,6 +24,8 @@ import (
 	"strings"
 	"time"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

Comment:
should be merged with the group below

Reply:
https://github.com/open-telemetry/opentelemetry-operator/pull/2666#issuecomment-2049075015

I'll work in another PR with log improvements since it is kind of expected behavior